src/Domain/Value/Branch.php

```diff
@@ -111,24 +119,32 @@ public function phpVersions(): array
     }
 
     /**
-     * @return Service[]
+     * @return Tool[]
+     */
+    public function tools(): array
+    {
+        return $this->tools;
+    }
+
+    /**
+     * @return string[]
      */
-    public function services(): array
+    public function phpExtensions(): array
     {
-        return $this->services;
+        return $this->phpExtensions;
     }
 
-    public function hasService(string $serviceName): bool
+    public function hasTool(string $toolName): bool
     {
-        if ([] === $this->services) {
+        if ([] === $this->tools) {
             return false;
         }
 
-        $serviceNames = array_map(static function (Service $service): string {
-            return $service->toString();
-        }, $this->services());
+        $toolNames = array_map(static function (Tool $tool): string {
+            return $tool->toString();
+        }, $this->tools());
```

Comment:
This way we can reduce useless checks and calls, since we only need to check for the first occurrence.
```suggestion
        foreach ($this->tools() as $tool) {
            if ($tool->toString() === $toolName) {
                return true;
            }
        }
        
        return false;
```

Reply:
Done.